fastddsspy_tool/test/application/test_class.py

```diff
@@ -0,0 +1,312 @@
+
+        self.logger.info('Executing command: ' + str(self.command))
+
+        proc = subprocess.Popen(self.command,
+                                stdin=subprocess.PIPE,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                encoding='utf8')
+
+        if (self.one_shot):
+            output = ''
+            try:
+                output = proc.communicate(timeout=10)[0]
+            except subprocess.TimeoutExpired:
+                proc.kill()
+            if not self.valid_output(output):
+                return ('wrong output')
+
+        else:
+            self.read_output(proc)
+        return proc
+
+    def send_command_tool(self, proc):
+        """
+        @brief Send a command to the running Spy.
+
+        @param proc: The subprocess object representing the running Spy.
+        @return Returns the output received after sending the command.
+        """
+        time.sleep(0.2)
+        proc.stdin.write((self.commands_spy[0]+'\n'))
+        proc.stdin.flush()
+
+        output = self.read_output(proc)
+        return (output)
+
+    def read_output(self, proc):
+        """
+        @brief Read the output from the subprocess.
+
+        @param proc: The subprocess object representing the running process.
+        @return Returns the accumulated output read from the subprocess.
+        """
+        output = ''
+        count = 0
+        max_count = 500
+
+        while True:
+            count += 1
+
+            if count > max_count:
+                break
+
+            line = proc.stdout.readline()
+
+            if ('Insert a command for Fast DDS Spy:' in line):
+                break
+
+            output = output + f'{line}\n'
+        return output
+
+    def output_command(self):
+        """
+        @brief Get the expected output.
+
+        @return Returns the expected output.
+        """
+        return (self.output)
+
+    def valid_guid(self, guid):
+        """
+        @brief Check if a GUID has the correct pattern.
+
+        @param guid: The GUID to check.
+        @return Returns True if the GUID is valid, False otherwise.
+        """
+        pattern = r'^((guid:)\s([0-9a-f]{2}\.){11}[0-9a-f]{2}\|([0-9a-f]\.){3}[0-9a-f]{1,})$'
+        id_guid = guid[guid.find('guid:'):]
+        if not re.match(pattern, id_guid):
+            return False
+        return True
+
+    def valid_rate(self, rate):
+        """
+        @brief Check if a rate is valid.
+
+        @param rate: The rate to check.
+        @return Returns True if the rate is valid, False otherwise.
+        """
+        pattern = r'^((rate:)\s\d{1,}\s(Hz))$'
+        id_rate = rate[rate.find('rate:'):]
+        if not re.match(pattern, id_rate):
+            return False
+        return True
+
+    def valid_output(self, output):
+        """Check the output ."""
+        if ('Fail' in self.name or (self.is_windows() and
```

Comment:
You have a class, use it. Do not infer information from a variable that is no forced to be related.
Also, do not under any circumstance add a specific test case condition in a general test (`--HelpCommand`).
```suggestion
        if (self.should_fail in self.name or (self.is_windows() and
```